### DISCUSSION TITLE
why llama.cpp inference with Vulkan backend by using Android GPU has very bad performance

### DISCUSSION THREAD

**Discussion by FranzKafkaYu** [ggml-org/llama.cpp#9464] [General]
related [issuse](https://github.com/ggerganov/llama.cpp/issues/8965)  

I have tried many times when I enable Vulkan backend with GPU acceleration,the performance is very bad.Currently I have tried Qulcomm Adreno GPU and ARM Mali GPU,Adreno GPU will faied in loading model,and Mali GPU can load model and execute inference but the Performance is very bad.  

I tried other projects,such as MLC-LLM and MediaPipe,they work with GPU and the performance is decent.Why llama.cpp can't compete with these projects?Can someone explain this to me？

**Comment by FranzKafkaYu** [comment]
with same model,same prompt,same output:  

reference with pure CPU will cost 1500ms~1700ms    

reference with  Vulkan(GPU) will cost 24000ms~25000ms

**Reply by kinchahoy** [reply]
+1 I see the same issue with Vulkan on ARM RK3588 devices. I don't think it's a driver issue.

**Reply by chrockjyy** [reply]
我在8gen3 adreno 750上用原生驱动也崩溃了，但是在termux里编译，使用turnip驱动可以加载成功，但是效率依然不高qwen3-8b-q4_k_m，7token左右

**Comment by amirvenus** [comment]
Has anyone managed to find a solution?

**Comment by SuperPauly** [comment]
Any updates?
